server/dive_server/viame_detection.py

```diff
@@ -217,37 +228,50 @@ def export_detections(self, folder, excludeBelowThreshold):
             default=False,
         )
     )
-    def export_all(self, folder, excludeBelowThreshold):
+    def export_all(
+        self, folder, includeMedia, includeDetections, excludeBelowThreshold
+    ):
         verify_dataset(folder)
         _, gen = self._generate_detections(folder, excludeBelowThreshold)
         setResponseHeader('Content-Type', 'application/zip')
         setContentDisposition(folder['name'] + '.zip')
         user = self.getCurrentUser()
         mediaFolder = getCloneRoot(user, folder)
+        source_type = fromMeta(mediaFolder, TypeMarker)
+        mediaRegex = None
+        if source_type == ImageSequenceType:
+            mediaRegex = imageRegex
+        elif source_type == VideoType:
+            mediaRegex = videoRegex
 
         def stream():
             z = ziputil.ZipGenerator(folder['name'])
-            # add media
-            for (path, file) in Folder().fileList(
-                mediaFolder,
-                user=user,
-                subpath=False,
-                mimeFilter=VideoMimeTypes.union(ImageMimeTypes),
-            ):
-                for data in z.addFile(file, path):
-                    yield data
-            # add JSON detections
-            for (path, file) in Folder().fileList(
-                folder,
-                user=user,
-                subpath=False,
-                mimeFilter={'application/json'},
-            ):
-                for data in z.addFile(file, path):
+
+            if includeMedia:
+                # Add media
+                for item in Folder().childItems(
+                    mediaFolder,
+                    filters={"lowerName": {"$regex": mediaRegex}},
+                ):
+                    for file in Item().childFiles(item, limit=1):
+                        path = File().getLocalFilePath(file)
```

Comment:
I don't think you want the local file path.  It returns the home/assetstore/ girder file folders instead of the girder folder paths.  You can use `Item().fileList` like above and stop after the first item.  There may be some special stuff I don't know about using that though.